openwisp_monitoring/monitoring/base/models.py

```diff
@@ -482,26 +483,68 @@ class AbstractAlertSettings(TimeStampedEditableModel):
     metric = models.OneToOneField(
         get_model_name('monitoring', 'Metric'), on_delete=models.CASCADE
     )
-    operator = models.CharField(max_length=1, choices=_ALERTSETTINGS_OPERATORS)
-    value = models.FloatField(help_text=_('threshold value'))
-    seconds = models.PositiveIntegerField(
-        default=0, validators=[MaxValueValidator(604800)], help_text=_(_SECONDS_HELP)
+    operator = models.CharField(
+        max_length=1, choices=_ALERTSETTINGS_OPERATORS, null=True, blank=True
+    )
+    threshold = models.FloatField(
+        _('threshold value'),
+        help_text=_(
+            'sets the threshold for collected metric data to send alerts, '
+            'you can set threshold value as a float'
+        ),
+        blank=True,
+        null=True,
+    )
+    tolerance = models.PositiveIntegerField(
+        _('threshold tolerance'),
+        default=0,
+        validators=[MaxValueValidator(_MINUTES_MAX)],
+        help_text=_(_MINUTES_HELP),
+        blank=True,
+        null=True,
     )
 
     class Meta:
         abstract = True
         verbose_name = _('Alert settings')
         verbose_name_plural = verbose_name
 
+    def full_clean(self, *args, **kwargs):
+        if self.threshold == self.config_dict['threshold']:
+            self.threshold = None
+        if self.tolerance == self.config_dict['tolerance']:
+            self.tolerance = None
+        if self.operator == self.config_dict['operator']:
+            self.operator = None
+        return super().full_clean(*args, **kwargs)
+
+    @property
+    def config_dict(self):
+        return self.metric.config_dict.get(
+            'alert_settings', {'operator': '<', 'threshold': 1, 'tolerance': 0}
+        )
+
+    @property
+    def get_threshold(self):
+        return self.threshold or self.config_dict['threshold']
+
+    @property
+    def get_tolerance(self):
+        return self.tolerance or self.config_dict['tolerance']
+
+    @property
+    def get_operator(self):
```

Comment:
methods named `get_` should not be decorated with `@property`.

I see mainly two possibilities:

- either leave these methods undecorated and call them like `instance.get_threshold()`
- rename the model fields as `custom_<field_name>` and rename these decorated methods to remove `get_` 

eg:

```python
@property
def tolerance(self):
    return self.custom_tolerance or self.config_dict['tolerance']
```